### ISSUE TITLE
Release 1.0 is coming!

### ISSUE THREAD

**Issue by fermadeiral** [kth-tcs/sonarqube-repair#49]
Hi @HarisAdzemovic, @nharrand, and @monperrus.

What do you think about creating a release with the current version of Sonarqube-repair? I thought about it because I think it's nice to register the end point in the project's history concerning the results reported in Haris' dissertation, which would be also the starting point of a new evolution round (of the tool and the data collection).

Cheers,
Fernanda

**Comment by HarisAdzemovic** [comment]
Hi,
Since contributing to open-source is so new to me I'll leave it up to you experts to decide. I'm not sure what a release usually implies though agree that this feels like a natural cut-off point in the project's history.

However, there is still some code left from the old system of creating processors (using the Sonarjava plugin) which has been stated in the README to no longer be maintained. Should that also be removed (or converted to the new system) before making a release?

**Comment by fermadeiral** [comment]
Hi @HarisAdzemovic,

> However, there is still some code left from the old system of creating processors (using the Sonarjava plugin) which has been stated in the README to no longer be maintained. Should that also be removed (or converted to the new system) before making a release?

I have noticed this situation too, and your question is really relevant. I plan to organize the repository as a whole before starting to merge into the master new features and new experimentation data. I will think on the situation and then I will come back to this issue.

Suggestions from the other contributors are more than welcome!

**Comment by fermadeiral** [comment]
Hi all,

I have worked in the past weeks to organize the repository before making the first release, which is coming in April 9th!

Well, since the tool is somewhat in the beginning of its development, I don't think that doing a detailed release notes for this first release is necessary (but we should do it in the next iterations), so the release notes for now will be the following:

# Release 1.0

## Handled rules

* Bug
    * Resources should be closed ([Sonar Rule 2095](https://rules.sonarsource.com/java/RSPEC-2095))
    * "BigDecimal(double)" should not be used ([Sonar Rule 2111](https://rules.sonarsource.com/java/RSPEC-2111))
    * "hashCode" and "toString" should not be called on array instances ([Sonar Rule 2116](https://rules.sonarsource.com/java/RSPEC-2116))
    * "Iterator.next()" methods should throw "NoSuchElementException" ([Sonar Rule 2272](https://rules.sonarsource.com/java/RSPEC-2272))
    * Strings and Boxed types should be compared using "equals()" ([Sonar Rule 4973](https://rules.sonarsource.com/java/RSPEC-4973))
* Code Smell
    * Unused assignments should be removed ([Sonar Rule 1854](https://rules.sonarsource.com/java/RSPEC-1854))
    * Fields in a "Serializable" class should either be transient or serializable ([Sonar Rule 1948](https://rules.sonarsource.com/java/RSPEC-1948))

**Comment by monperrus** [comment]
Waooowww, nice!

**Comment by HarisAdzemovic** [comment]
Magnificent work, @fermadeiral!

**Comment by fermadeiral** [comment]
We are good, boys! We have a first GitHub release for this project: https://github.com/kth-tcs/sonarqube-repair/releases/tag/v1.0

**Comment by monperrus** [comment]
Thanks a lot Fernanda.
